pkg/probe/http/http.go

```diff
@@ -28,15 +29,18 @@ import (
 )
 
 func New() HTTPProber {
-	return HTTPProber{&http.Client{}}
+	client := &http.Client{}
+	return HTTPProber{client, func(timeout time.Duration) { client.Timeout = timeout }}
 }
 
 type HTTPProber struct {
-	client HTTPGetInterface
+	client     HTTPGetInterface
+	setTimeout func(timeout time.Duration)
 }
 
 // Probe returns a ProbeRunner capable of running an http check.
-func (pr *HTTPProber) Probe(host string, port int, path string) (probe.Status, error) {
+func (pr *HTTPProber) Probe(host string, port int, path string, timeout time.Duration) (probe.Status, error) {
+	pr.setTimeout(timeout)
```

Comment:
Is client.Do guaranteed to return? I suppose the default redirect policy would likely guarantee that. If not, doesn't that potentially result in a goroutine leak similar to the prior Exec timeout handling?